benchmarks/base.py

```diff
@@ -18,45 +21,66 @@ class BaseBench:
     processes = 1
     timeout = 300
 
-    def setup(self):
+    def setup(self, *params):
         self.cwd = os.getcwd()
         self.test_directory = TemporaryDirectory(prefix="DVCBenchmark")
-        os.chdir(self.test_directory.name)
+        os.chdir(self.path)
+
+    @property
+    def path(self):
+        return self.test_directory.name
 
-    def teardown(self):
+    def teardown(self, *params):
         os.chdir(self.cwd)
 
         if hasattr(self, "repo"):
             self.repo.close()
 
         if os.name == "nt":
             # Windows does not allow to remove read only files
-            for root, _, files in os.walk(self.test_directory.name):
+            for root, _, files in os.walk(self.path):
                 for file in files:
                     path = os.path.join(root, file)
                     perm = os.stat(path).st_mode | stat.S_IWRITE
                     os.chmod(path, perm)
 
         self.test_directory.cleanup()
 
-
-def init_git(path):
-    from git import Repo
-
-    git = Repo.init(path)
-    git.close()
-
-
-def init_dvc(path, git=True):
-    from dvc.repo import Repo
-
-    if git:
-        init_git(path)
-        repo = Repo.init(path)
-        repo.scm.commit("Init DVC repo")
-    else:
-        repo = Repo.init(path, no_scm=True).close()
-    return repo
+    def dvc(self, *args, return_code=0, proc=False):
+        if proc:
+            proc = Popen(["dvc", *args], stdout=PIPE)
+            proc.communicate()
+            assert proc.returncode == return_code
+        else:
+            assert main(args) == return_code
+
+    def init_git(self):
+        from git import Repo
+
+        Repo.init(self.path).close()
+
+    def init_dvc(self, no_scm=False):
+        from dvc.repo import Repo
+
+        if not no_scm:
+            self.init_git()
+            repo = Repo.init(self.path)
+            repo.scm.commit("Init DVC repo")
+        else:
+            repo = Repo.init(self.path, no_scm=True)
+        repo.close()
```

Comment:
Let's decouple dvc from git, same as we have in our fixtures in dvc. So to get dvc with git you would:
```
self.init_git()
self.init_dvc()
```